Source/lua/modules/items.cpp

```diff
@@ -18,98 +19,94 @@ void InitItemUserType(sol::state_view &lua)
+	LuaSetDocProperty(itemType, "postDraw", "boolean", "Post-draw flag", &Item::_iPostDraw);
+	LuaSetDocProperty(itemType, "identified", "boolean", "Identified flag", &Item::_iIdentified);
+	LuaSetDocProperty(itemType, "magical", "number", "Item quality", [](const Item &i) { return static_cast<int>(i._iMagical); }, [](Item &i, int val) { i._iMagical = static_cast<item_quality>(val); });
+	LuaSetDocProperty(itemType, "name", "string", "Item name", [](const Item &i) { return std::string(i._iName); }, [](Item &i, const std::string &val) { CopyUtf8(i._iName, val, sizeof(i._iName)); });
+	LuaSetDocProperty(itemType, "iName", "string", "Identified item name", [](const Item &i) { return std::string(i._iIName); }, [](Item &i, const std::string &val) { CopyUtf8(i._iIName, val, sizeof(i._iIName)); });
+	LuaSetDocProperty(itemType, "loc", "ItemEquipType", "Equipment location", [](const Item &i) { return i._iLoc; }, [](Item &i, int val) { i._iLoc = static_cast<item_equip_type>(val); });
+	LuaSetDocProperty(itemType, "class", "ItemClass", "Item class", [](const Item &i) { return i._iClass; }, [](Item &i, int val) { i._iClass = static_cast<item_class>(val); });
+	LuaSetDocProperty(itemType, "curs", "number", "Cursor index", &Item::_iCurs);
+	LuaSetDocProperty(itemType, "value", "number", "Item value", &Item::_ivalue);
+	LuaSetDocProperty(itemType, "ivalue", "number", "Identified item value", &Item::_iIvalue);
+	LuaSetDocProperty(itemType, "minDam", "number", "Minimum damage", &Item::_iMinDam);
+	LuaSetDocProperty(itemType, "maxDam", "number", "Maximum damage", &Item::_iMaxDam);
+	LuaSetDocProperty(itemType, "AC", "number", "Armor class", &Item::_iAC);
+	LuaSetDocProperty(itemType, "flags", "ItemSpecialEffect", "Special effect flags", [](const Item &i) { return static_cast<int>(i._iFlags); }, [](Item &i, int val) { i._iFlags = static_cast<ItemSpecialEffect>(val); });
+	LuaSetDocProperty(itemType, "miscId", "ItemMiscID", "Miscellaneous ID", [](const Item &i) { return i._iMiscId; }, [](Item &i, int val) { i._iMiscId = static_cast<item_misc_id>(val); });
+	LuaSetDocProperty(itemType, "spell", "SpellID", "Spell", [](const Item &i) { return i._iSpell; }, [](Item &i, int val) { i._iSpell = static_cast<SpellID>(val); });
+	LuaSetDocProperty(itemType, "IDidx", "ItemIndex", "Base item index", [](const Item &i) { return i.IDidx; }, [](Item &i, int val) { i.IDidx = static_cast<_item_indexes>(val); });
+	LuaSetDocProperty(itemType, "charges", "number", "Number of charges", &Item::_iCharges);
+	LuaSetDocProperty(itemType, "maxCharges", "number", "Maximum charges", &Item::_iMaxCharges);
+	LuaSetDocProperty(itemType, "durability", "number", "Durability", &Item::_iDurability);
+	LuaSetDocProperty(itemType, "maxDur", "number", "Maximum durability", &Item::_iMaxDur);

```

Comment:
Unrelated, but are all these Lua APIs public already? Are there plans to cleanup the inconsistent naming conventions?

Reply:
@Iniquitatis the goal for 1.6.0 is to largely decouple Hellfire from the engine and turn as many QoL as makes sens in to Lua mods. Mostly the status of all of this is that data tables have been extracted in to .tsv and MPQ files can now be enabled / disabled from an in game Mods menu, and debug commands have been moved to a Lua based console.

Mods can also include a Lua script that gets loaded from there MPQ.

Lastly we also want to expose enough that we can write a Lua based accessibility mod with it.